pandas/io/stata.py

```diff
@@ -1979,11 +1979,16 @@ def _open_file_binary_write(
         compression_typ, compression_args = get_compression_method(compression)
         compression_typ = infer_compression(fname, compression_typ)
         compression = dict(compression_args, method=compression_typ)
-        path_or_buf, _, compression, _ = get_filepath_or_buffer(
+        ioargs = get_filepath_or_buffer(
             fname, mode="wb", compression=compression, storage_options=storage_options,
         )
-        f, _ = get_handle(path_or_buf, "wb", compression=compression, is_text=False)
-        return f, True, compression
+        f, _ = get_handle(
```

Comment:
I assume the auxiliary file handles should be closed as well? I think that only happens for compressed files. But that would require yet another interface change.